containers/ecr-viewer/src/app/components/BaseFilter.tsx

```diff
@@ -329,3 +330,108 @@ export const CustomDateInput = ({
     </div>
   );
 };
+
+/**
+ * Checkbox to Select/Deselect all depending on checked status.
+ * @param props - The properties for the Select/Deselect all checkbox.
+ * @param props.groupName - The name of the group that's being selected/deselected.
+ * @param props.onToggle - The callback function to handle the toggle event when the checkbox is clicked.
+ * @param props.isAllSelected - Indicates whether all are selected.
+ * @returns The rendered Select/Deselect all checkbox component.
+ */
+export const SelectDeselectAllCheckbox = ({
+  groupName,
+  onToggle,
+  isAllSelected,
+}: {
+  groupName: string;
+  onToggle: () => void;
+  isAllSelected: boolean;
+}) => {
+  return (
+    <CheckboxInput
+      id={`${groupName}-all`}
+      name={isAllSelected ? "Deselect all" : "Select all"}
+      value="all"
+      checked={isAllSelected}
+      onChange={() => onToggle()}
+      classNamesDiv="padding-bottom-1 padding-x-105"
+      classNamesLabel="line-height-sans-6 font-sans-xs margin-y-0"
+    />
+  );
+};
+
+/**
+ * A group of checkbox button components, given a set of options.
+ * @param props - The properties for the Checkbox group component.
+ * @param props.groupName - The name of the checkbox buttons group.
+ * @param props.filterItems - The option currently selected.
+ * @param props.onChange - The callback function to handle the `onChange` event when the checkbox is clicked.
+ * @returns The rendered checkbox group component.
+ */
+export const CheckboxOptions = ({
+  groupName,
+  filterItems,
+  onChange,
+}: {
+  groupName: string;
+  filterItems: { [key: string]: boolean };
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}) => {
+  return (
+    <div className="position-relative bg-white overflow-y-auto maxh-38 display-flex flex-column gap-1 padding-y-1 padding-x-105">
+      {Object.keys(filterItems).map((item) => (
+        <CheckboxInput
+          key={item}
+          id={`${groupName}-${item}`}
+          name={item}
+          value={item}
+          checked={filterItems[item]}
+          onChange={onChange}
+          classNamesLabel="line-height-sans-6 font-sans-xs margin-y-0 minw-40"
+        />
+      ))}
+    </div>
+  );
+};
+
+type CheckboxInputProps = {
```

Comment:
could you add a comment as to why we're using this vs the standard react-uswds component?